### PR TITLE
feat(datepicker): reworked datepicker and calendar design to match the material specs

### LINKED ISSUE #4251: md-datepicker should have option to select year and month
If we have to select future of past dates, (date of birth of something), it is difficult to scroll up and down. Should have selection of year and month.

### LINKED ISSUE #5779: Select year support in md-datepicker
screen shot 2015-11-17 at 1 39 28 pm

### PR DESCRIPTION
**Pull request by DevVersion**
## General
This is a second PR from me, which redesigns completely the datepicker.
The other PR only adds the Month Selector (so this PR contains this PR too).
I made this, so you only need to decide which PR you take.

**Expected Result by Google Material Specs**

## Showcase
**Default View**

**Use the Chevrons**

**Scrolling is updating the Month Selector**

## Important Information
PR Depends on #5967

## At last
If there are any suggestions or wishes, I'm always up for that :smile: 

 Closes #5967 Fixes #5779 Fixes #4251

### CODE REVIEW

**Comment by EladBezalel** [comment]
@DevVersion thank you for your contribution! 
But, as much as it seems nice, the spec comparison is wrong. 
The spec specifies a <b>Dialog</b> datepicker and not a menu datepicker.
i guess @jelbourn who wrote this component have a different design handled directly from the UX team.

**Comment by DevVersion** [comment]
That's right, but it should at least look like a Google Material Component, this PR has the same features as before + it contains the calendar month selector + datepicker header

So it is only a improvement for the UX

**Comment by EladBezalel** [comment]
As you can see [here](https://material-design.storage.googleapis.com/publish/material_v_4/material_ext_publish/0B6Okdz75tqQsSm5KN1FNQXRFRDA/components_pickers_date_navigation_xhdpi_004.webm) dialog is behaving different from segmented dropdown button, IMO we shouldn't guess and ask the UX team.

**Comment by DevVersion** [comment]
I'm currently working on the horizontal scroll. And yes that's a good idea. But IMO it looks more Material (as in the specs for dialog - but that should be still better than an unknown design).

Who is the UX Team (I know nothing about that :+1:)?

**Comment by Soulcyber** [comment]
is there a fast year selection in your picker?
thanks for the work :)

**Comment by DevVersion** [comment]
No not yet, I will try to inplemenent that If I got time.

**Comment by jelbourn** [comment]
@DevVersion Thanks for your contribution, but we aren't going to be changing the appearance of the datepicker. We are following a specification from our internal UX people that hasn't been published to the Material Design site yet (but hopefully will soon). As far as a month selector goes, @mhchen is actively working on adding such a view that follows the aforementioned spec.

We love contributions, but please check with us (myself or @ThomasBurleson) before diving into any major work, as we can usually steer you in the right direction WRT our plans for the library.

File: src/components/datepicker/calendar.js [hunk 1]
```diff
@@ -217,6 +234,13 @@
       }
     };
 
+    /**
+     * Fire watchers of scope on scrolling to refresh the current month label
+     */
+    this.calendarScroller.addEventListener('scroll', function() {
+      $scope.$digest();
```
Comment (jelbourn) [inline src/components/datepicker/calendar.js]: FYI: doing this would completely annihilate application performance.

**Comment by DevVersion** [comment]
Hello @jelbourn,

I happy to get a response from you :+1: As always I'm just doing this as an hobby so it isn't bad for me :smile: I can completely accept your choice and from the UX Team. 

And about checking with you, I don't know how to get in contact (as a chat) with you. But that's not the problem :smile:. I'm always trying to made changes with your decisions. So I only try to give as much possibilities for you (so you can pick the best / or wanted).

So thanks again for your response. I'm looking forward to see the internal specifications.

**Pull request closed without merging**